src/compiler/checker.ts

```diff
@@ -11987,19 +11994,19 @@ namespace ts {
                         // If all inferences were made from contravariant positions, infer a common subtype. Otherwise, if
                         // union types were requested or if all inferences were made from the return type position, infer a
                         // union type. Otherwise, infer a common supertype.
-                        const unwidenedType = context.flags & InferenceFlags.InferUnionTypes || inference.priority & InferencePriority.PriorityImpliesUnion ?
+                        const unwidenedType = context.flags & InferenceFlags.InferUnionTypes || inference.priority & InferencePriority.PriorityImpliesCombination ?

```

Comment:
can you extract this into a function similar to `getContravariantInference`? Or naming it something besides `unwidenedType` would make it more obvious why we  check PriorityImpliesCombination and get the union, but then get the intersection if that type is never.

Reply:
Looks like it is #22323

Reply:
@sandersn how is it now?